File: 프로그래머스/1/135808. 과일 장수/과일 장수.py
def solution(k, m, score):
    answer = 0
    array = []
    #사과상자 가격 = 낮은점수(p) x 사과의개수(m)
    score.sort(reverse=True)
    for i in range(0, len(score), m):
        if len(score[i:i+m]) == m:
            answer += min(score[i:i+m]) * m
        else:
            break
    
    return answer
    

File: 프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        
        Arrays.sort(score);

        for(int i=0; i<score.length/2; i++) {
            int temp = score[i];
            score[i] = score[score.length-1-i];
            score[score.length-1-i] = temp;
        }
        
        for(int i=0; i<score.length-m+1; i+=m) {
            answer += score[i+m-1] * m;
        }
        
        return answer;
    }
}